### PR TITLE
Nodejs tools

### PR DESCRIPTION
**Pull request by airportyh**
## Why?

We want Nix module support for Node.js.

## Changes

Created Node.js module, which contains:

1. nodejs
2. prybar-nodejs
3. typescript language server
4. js-debug. To package js-debug, we load [our fork of js-debug](https://github.com/replit/vscode-js-debug) and run some gulp commands. Node2nix is used and is used to generate the `.nix` files in js-debug with the package.json as input.

### CODE REVIEW

**Review by cdmistman — APPROVED** [review]
lfggggg

File: pkgs/dapNode/js-debug/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# js-debug Package
+
+This Nix package installs and builds the DAP server in
+https://github.com/microsoft/vscode-js-debug.
```
Comment (cdmistman) [inline pkgs/dapNode/js-debug/README.md:4]: isn't it
```suggestion
https://github.com/replit/vscode-js-debug.
```

File: pkgs/dapNode/js-debug/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+The `.nix` files were generated using [node2nix](https://github.com/svanderburg/node2nix) by the command:
+
+```
+node2nix -d
+```
+
+which uses the `package.json` file as input, which was grabbed from https://github.com/svanderburg/node2nix
+at rev 483425e50cf0fb313610d2346b51459c8db556e2, then removing the playwright package as I couldn't
```
Comment (cdmistman) [inline pkgs/dapNode/js-debug/README.md:13]: is the rev for node2nix or for vscode-js-debug? if it's for the former, isn't that the version that's pinned in this repo? if not, can it be added to the devshell?

File: modules/nodejs.nix [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+in
+
+{
+  name = "Node.js Tools";
+  version = "1.0";
+
+  packages = [
+    nodejs
+    typescript-language-server
```
Comment (cdmistman) [inline modules/nodejs.nix:32]: can't this be implicitly used in `languageServers`? not sure why it needs to be included in `packages` - also plays nicer with deployments

File: pkgs/dapNode/js-debug/README.md (hunk 1 above)
Comment (airportyh) [reply pkgs/dapNode/js-debug/README.md:4]: :+1:

**Review by ryantm — APPROVED** [review]
looks good, I didn't try it though.

**Pull request merged**